FunctionBar.c

```diff
@@ -92,6 +92,28 @@ int FunctionBar_draw(const FunctionBar* this) {
    return FunctionBar_drawExtra(this, NULL, -1, false);
 }
 
+static size_t FunctionBar_displayWidth(const char *s) {
+   size_t slen = strlen(s);
+   size_t width = 0, i = 0;
+
+   mbtowc(NULL, 0, 0);
+
+   while (i < slen) {
+      wchar_t wch;
+      int mblen = mbtowc(&wch, s + i, slen - i);
```

Comment:
Ditto. (Use `mbrtowc` here rather than downgrade to `mbtowc`. With `mbrtowc`, the string length can be properly addressed as `size_t` type.)